src/main/kotlin/com/dansplugins/factionsystem/gate/MfGateService.kt

```diff
@@ -56,9 +56,21 @@ class MfGateService(
     fun getGatesByStatus(status: MfGateStatus) = gatesById.values.filter { it.status == status }
 
     fun save(gate: MfGate) = resultFrom {
-        val result = gateRepo.upsert(gate)
-        gatesById[result.id] = result
-        return@resultFrom result
+        try {
+            val result = gateRepo.upsert(gate)
+            gatesById[result.id] = result
+            return@resultFrom result
+        } catch (e: OptimisticLockingFailureException) {
+            plugin.logger.warning("Optimistic locking failure for gate with ID: ${gate.id}. Retrying...")
+            val reloadedGate = gateRepo.getGate(gate.id) ?: throw IllegalStateException("Gate not found: ${gate.id}")
+            val mergedGate = reloadedGate.copy(
+                status = gate.status,
+                version = reloadedGate.version + 1
```

Comment:
Version should be updated in the SQL so you shouldn't need to touch it in code: https://github.com/Dans-Plugins/Medieval-Factions/blob/main/src/main/kotlin/com/dansplugins/factionsystem/gate/JooqMfGateRepository.kt#L62

When you fetch the new gate you should have the most recent version so it should avoid the failure.